meson.build

```
@@ -0,0 +1,376 @@
+    'sparc',
+    'riscv',
+]
+
+simple_filters = [
+    'x86',
+    'powerpc',
+    'ia64',
+    'arm',
+    'armthumb',
+    'arm64',
+    'sparc',
+    'riscv',
+]
+
+# FIXME foreach i in array; set_variable(f'have_@i@', true) endforeach
+
+foreach f : supported_filters
+    if get_option('encoders').contains(f)
+        config_h.set10('HAVE_ENCODER_' + f.underscorify().to_upper(), true,
+            description: 'Define to 1 if ' + f + ' encoder is enabled.'
+        )
+        encoder_msg += f + ' '
+    endif
+    if get_option('decoders').contains(f)
+        config_h.set10('HAVE_DECODER_' + f.underscorify().to_upper(), true,
+            description: 'Define to 1 if ' + f + ' decoder is enabled.'
+        )
+        decoder_msg += f + ' '
+    endif
+endforeach
+
+have_encoder_simple_filters = false
+have_decoder_simple_filters = false
+foreach f : simple_filters
+    if get_option('encoders').contains(f)
+        have_encoder_simple_filters = true
+    endif
+    if get_option('decoders').contains(f)
+        have_decoder_simple_filters = true
+    endif
+endforeach
+
+if (get_option('encoders').contains('lzma2') and not get_option('encoders').contains('lzma1')) or (get_option('decoders').contains('lzma2') and not get_option('decoders').contains('lzma1'))
+    error('LZMA2 requires that LZMA1 is also enabled')
+endif
+
+if get_option('encoders').length() > 0
+    config_h.set10('HAVE_ENCODERS', true,
+        description: 'Define to 1 if at least one encoder has been enabled.'
+    )
+endif
+
+if get_option('decoders').length() > 0
+    config_h.set10('HAVE_DECODERS', true,
+        description: 'Define to 1 if at least one decoder has been enabled.'
+    )
+endif
+
+### Match finders
+
+match_finder_msg = ''
+
+match_finders = [
+    'hc3',
+    'hc4',
+    'bt2',
+    'bt3',
+    'bt4'
+]
+
+if get_option('encoders').contains('lzma2') == true or get_option('encoders').contains('lzma1') == true
+    if get_option('match-finders').length() == 0
+        error('at least one match finder is required for an LZ-based encoder')
+    endif
+
+    foreach m : match_finders
+        if get_option('match-finders').contains(m) == true
+            config_h.set10('HAVE_MF_' + m.underscorify().to_upper(), true,
+                description: 'Define to 1 to enable ' + m + ' match finder.'
+            )
+            match_finder_msg += m + ' '
+        endif
+    endforeach
+endif
+
+### Integrity checks
+
+check_msg = ''
+
+checks = [
+    'crc32',
+    'crc64',
+    'sha256'
+]
+
+foreach c : checks
+    if get_option('checks').contains(c) == true
+        config_h.set10('HAVE_CHECK_' + c.underscorify().to_upper(), true,
+            description: 'Define to 1 if ' + c + ' integrity check is enabled.'
+        )
+        check_msg += c + ' '
+    endif
+endforeach
+
+if get_option('checks').contains('crc32') == false
+    error('For now, the CRC32 check must always be enabled')
+endif
+
+### microLZMA
+
+if get_option('microlzma') == true
```

```suggestion
if get_option('microlzma')
```

All `== true` comparisons are redundant.

There are dozens of instances of this sort of issue. This is a really weird pattern that difficult to read, you're basically doing true == true in every case, because Meson does not allow type coercion to `bool` (not even from `int` to `bool`), nor does it allow comparisons between different types.

This isn't C :wink:

---

i'm too used to C indeed :-)
done in 72f696e5426cfa3e6ee9e294108e051d39890253